pkg/bee/client.go

```diff
@@ -38,14 +41,17 @@ type ClientOptions struct {
 	APIURL         *url.URL
 	Name           string
 	Retry          int
+	SwapClient     swap.Client
 }
 
 // NewClient returns Bee client
 func NewClient(opts ClientOptions, log logging.Logger) (c *Client) {
 	c = &Client{
-		retry: retryCount,
-		opts:  opts,
-		log:   log,
+		retry:      retryCount,
+		log:        log,
+		swapClient: opts.SwapClient,
```

Comment:
In case that opts.SwapClient is nil, use swap.NotSet.

Reply:
thanks for the idea :)